packages/block-library/src/image/index.php

```diff
@@ -32,44 +32,46 @@ function render_block_core_image( $attributes, $content, $block ) {
 		$processor->set_attribute( 'data-id', $attributes['data-id'] );
 	}
 
-	$lightbox_enabled  = false;
 	$link_destination  = isset( $attributes['linkDestination'] ) ? $attributes['linkDestination'] : 'none';
 	$lightbox_settings = block_core_image_get_lightbox_settings( $block->parsed_block );
 
-	// If the lightbox is enabled and the image is not linked, flag the lightbox to be rendered.
-	if ( isset( $lightbox_settings ) && 'none' === $link_destination ) {
-
-		if ( isset( $lightbox_settings['enabled'] ) && true === $lightbox_settings['enabled'] ) {
-			$lightbox_enabled = true;
-		}
+	$view_js_file_handle = 'wp-block-image-view';
+	$script_handles      = $block->block_type->view_script_handles;
+
+	/*
+	 * Remove the filter and the JavaScript view file if previously added by
+	 * other Image blocks.
+	 */
+	remove_filter( 'render_block_core/image', 'block_core_image_render_lightbox', 15 );
+	// If the script is not needed, and it is still in the `view_script_handles`, remove it.
+	if ( in_array( $view_js_file_handle, $script_handles, true ) ) {
+		$block->block_type->view_script_handles = array_diff( $script_handles, array( $view_js_file_handle ) );
 	}
 
-	// If at least one block in the page has the lightbox, mark the block type as interactive.
-	if ( $lightbox_enabled ) {
+	/*
+	 * If the lightbox is enabled and the image is not linked, add the filter
+	 * and the JavaScript view file.
+	 */
+	if (
+		isset( $lightbox_settings ) &&
+		'none' === $link_destination &&
+		isset( $lightbox_settings['enabled'] ) &&
+		true === $lightbox_settings['enabled']
+	) {
 		$block->block_type->supports['interactivity'] = true;
-	}
-
-	// Determine whether the view script should be enqueued or not.
-	$view_js_file = 'wp-block-image-view';
-	if ( ! wp_script_is( $view_js_file ) ) {
-		$script_handles = $block->block_type->view_script_handles;
 
-		// If the script is not needed, and it is still in the `view_script_handles`, remove it.
-		if ( ! $lightbox_enabled && in_array( $view_js_file, $script_handles, true ) ) {
-			$block->block_type->view_script_handles = array_diff( $script_handles, array( $view_js_file ) );
+		if ( ! in_array( $view_js_file_handle, $script_handles, true ) ) {
```

Comment:
It appears that when running this code when the lightbox is enabled for just the first image block, `$script_handles` contains the `'wp-block-image-view'` handle, so this conditional erroneously evaluates to `false` and the view script is not loaded, resulting in the following error:

<img width="1032" alt="Screenshot 2023-10-06 at 4 36 22 PM" src="https://github.com/WordPress/gutenberg/assets/5360536/e9d3a7ab-e512-4821-b171-774bd8227dc4">